hi-ml-cpath/src/health_cpath/preprocessing/tiff_conversion.py

```diff
@@ -0,0 +1,241 @@
+        :param image_key: The key of the image in the data dictionary, defaults to SlideKey.IMAGE
+        :param src_format: The format of the src file, defaults to WSIFormat.NDPI
+        :param target_magnifications: The target magnifications e.g. [10., 20.], defaults to [10.]. If
+        target_magnifications is None, the tiff file will contain the image data at all magnifications.
+        :param add_lowest_magnification: A flag indicating whether the tiff file should also contain the image data at
+        the lowest magnification, defaults to False. This is useful if the lowest magnification of the wsi is not part
+        of the target magnifications and one wants to use the lowest magnification for faster processing.
+        :param base_objective_power: The base objective power of the wsi. This is used to calculate the magnification
+        of the wsi. If the objective power is not found in the wsi properties, the base_objective_power is used
+        instead. If the objective power is not found in the wsi properties and base_objective_power is None, an error
+        is raised., defaults to None
+        :param replace_ampersand_by: A string that is used to replace ampersands in the src file name, defaults to
+        UNDERSCORE. This is useful because ampersands in file names can cause problems in cloud storage.
+        :param compression: The compression that is used to save the tiff file, defaults to COMPRESSION.ADOBE_DEFLATE
+        aka ZLIB that is lossless compression.
+        :param tile_size: The size of the tiles that are used to write the tiff file, defaults to 512.
+        :param backend: The backend that is used to read the wsi file, defaults to WSIBackend. OPENSLIDE which is
+        compatible with most wsi formats.
+        """
+        self.output_folder = output_folder
+        self.image_key = image_key
+        self.src_format = src_format
+        if target_magnifications is not None:
+            target_magnifications.sort(reverse=True)
+        self.target_magnifications = target_magnifications if target_magnifications else None
+        self.add_lowest_magnification = add_lowest_magnification
+        self.replace_ampersand_by = replace_ampersand_by
+        self.base_objective_power = base_objective_power
+        self.wsi_reader = WSIReader(backendbackend=backend)
+        self.compression = compression
+        self.tile_size = tile_size
+
+    def get_tiff_path(self, src_path: Path) -> Path:
+        """Returns the path to the tiff file that will be created from the src file. The tiff file is saved in the
+        output_folder with the same name as the src file but with the tiff extension. Ampersands are replaced by the
+        replace_ampersand_by string.
+
+        :param src_path: The path to the src file.
+        :return: The path to the tiff file that will be created from the src file.
+        """
+        tiff_filename = src_path.name.replace(self.src_format, WSIFormat.TIFF)
```

Comment:
You only want to change the extension/suffix, so it is safer to use `Path.with_suffix` https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.with_suffix

Reply:
oh didn't know about this, it's a lot nice, thanks!